lib/demos/material/navigation_rail_demo.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/gallery_localizations.dart';
+
+// BEGIN navRailDemo
+
+class NavRailDemo extends StatefulWidget {
+  const NavRailDemo({Key key}) : super(key: key);
+
+  @override
+  _NavRailDemoState createState() => _NavRailDemoState();
+}
+
+class _NavRailDemoState extends State<NavRailDemo> {
+  int _selectedIndex = 0;
+
+  @override
+  Widget build(BuildContext context) {
+    final localization = GalleryLocalizations.of(context);
+    final demoNavigationRailFirst = localization.demoNavigationRailFirst;
+    final demoNavigationRailSecond = localization.demoNavigationRailSecond;
+    final demoNavigationRailThird = localization.demoNavigationRailThird;
+    final selectedItem = <String>[
+      demoNavigationRailFirst,
+      demoNavigationRailSecond,
+      demoNavigationRailThird
+    ];
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(
+          localization.demoNavigationRailTitle,
+        ),
+      ),
+      body: Row(
+        children: [
+          NavigationRail(
+            selectedIndex: _selectedIndex,
```

Comment:
I think it would be nice to show `leading` in use here with a `FloatingActionButton` since that is a common case:

https://material.io/components/navigation-rail

Reply:
Good idea, I'll add it.